src/internal/m365/collection/groups/metadata.go

```diff
@@ -0,0 +1,130 @@
+package groups
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/pkg/backup/metadata"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// ParseMetadataCollections produces a map of structs holding delta
+// and path lookup maps.
+func parseMetadataCollections(
```

Comment:
I guess we might have to tweak this depending on how https://github.com/alcionai/corso/pull/4205/files#diff-b04249345cf488fc7afc745f109b597739999c29a9b4325f8e31ff3fcc400c41R82 works out.

Reply:
Possibly, yeah.  We'll consider this a pin on the idea, and we'll know to follow-up on it.